src/Symfony/Bridge/Twig/Node/SearchAndRenderBlockNode.php

```diff
@@ -24,88 +28,92 @@ class SearchAndRenderBlockNode extends FunctionExpression
     public function compile(Compiler $compiler)
     {
         $compiler->addDebugInfo($this);
-        $compiler->raw('$this->env->getRuntime(\'Symfony\Component\Form\FormRenderer\')->searchAndRenderBlock(');
-
         preg_match('/_([^_]+)$/', $this->getAttribute('name'), $matches);
-
         $label = null;
-        $arguments = iterator_to_array($this->getNode('arguments'));
         $blockNameSuffix = $matches[1];
 
-        if (isset($arguments[0])) {
-            $compiler->subcompile($arguments[0]);
-            $compiler->raw(', \''.$blockNameSuffix.'\'');
-
+        $arguments = iterator_to_array($this->getNode('arguments'));
+        if ('label' === $blockNameSuffix) {
+            if (isset($arguments[2])) {
+                $arguments['variables'] = $arguments[2];
+                unset($arguments[2]);
+            }
+            $label = null;
             if (isset($arguments[1])) {
-                if ('label' === $blockNameSuffix) {
-                    // The "label" function expects the label in the second and
-                    // the variables in the third argument
-                    $label = $arguments[1];
-                    $variables = isset($arguments[2]) ? $arguments[2] : null;
-                    $lineno = $label->getTemplateLine();
-
-                    if ($label instanceof ConstantExpression) {
-                        // If the label argument is given as a constant, we can either
-                        // strip it away if it is empty, or integrate it into the array
-                        // of variables at compile time.
-                        $labelIsExpression = false;
-
-                        // Only insert the label into the array if it is not empty
-                        if (!twig_test_empty($label->getAttribute('value'))) {
-                            $originalVariables = $variables;
-                            $variables = new ArrayExpression(array(), $lineno);
-                            $labelKey = new ConstantExpression('label', $lineno);
+                $label = $arguments[1];
+                unset($arguments[1]);
+            } elseif (isset($arguments[1])) {
```

Comment:
Shouldn't this be `isset($arguments['label'])` ? `isset($arguments['label'])` is already checked earlier in the code.